drivers/hmis/app/models/hmis/hud/processors/referral_request_processor.rb

```diff
@@ -16,5 +16,15 @@ def schema
 
     def information_date(_)
     end
+
+    def assign_metadata
+      record = @processor.send(factory_name)
+      return if record.persisted?
+
+      record.assign_attributes(
+        requested_by: @processor.current_user,
+        identifier: SecureRandom.uuid, # FIXME we should be calling an external endpoint
+      )
+    end
```

Comment:
@ttoomey heads up about this - is it alright? I moved the logic of assigning the "user" and any "metadata" into this assign_metadata func on the processors. SubmitForm is simplified.

Reply:
Got it, removed